src/pages/EditShelterSupply/EditShelterSupply.tsx

```diff
@@ -100,13 +104,31 @@ const EditShelterSupply = () => {
         },
       });
     },
-    [refresh, shelterId, toast]
+    [location.state, navigate, refresh, shelterId, toast]
   );
 
   useEffect(() => {
     setFilteredSupplies(supplies);
   }, [supplies]);
 
+  useEffect(() => {
+    const { supplyIdToAdd } = location.state ?? {}
+    
+    if (!supplyIdToAdd) return
+
+    const findShelterDataSupply = shelterSupplyData[supplyIdToAdd]
+
+    if (findShelterDataSupply && findShelterDataSupply.supply) {
+      const { supply, priority } = findShelterDataSupply
+
+      handleClickSupplyRow({
+        id: supply?.id,
+        name: supply?.name,
+        priority
+      }) 
+    }
```

Comment:
Aqui ao invés de dar o return, talvez fosse melhor apenas só executar o restante se o `supplyIdToAdd` for válido, já que o useEffect espera que seja retornada uma função de cleanup.
```suggestion
    if (supplyIdToAdd) {
      const findShelterDataSupply = shelterSupplyData[supplyIdToAdd]

      if (findShelterDataSupply && findShelterDataSupply.supply) {
        const { supply, priority } = findShelterDataSupply

        handleClickSupplyRow({
          id: supply?.id,
          name: supply?.name,
          priority
        }) 
      }
    }
```